# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build libpng-build sdl-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

pkg-config --exists jpeg-build && REQUIRES="$REQUIRES jpeg-build"
pkg-config --exists libwebp-build && REQUIRES="$REQUIRES libwebp-build"
pkg-config --exists tiff-build && REQUIRES="$REQUIRES tiff-build"

# configure (ac-2.61; am-1.10; lt-2.2.6)
libtoolize -c -f; aclocal -I acinclude; autoconf; automake -a -c
PKG_CONFIG_PATH=/x11/lib/pkgconfig LDFLAGS=-Wl,-rpath,/x11/lib ./configure --disable-static --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/bin
install .libs/showimage $DESTDIR/x11/bin
install -d $DESTDIR/x11/include/SDL
install -m 644 SDL_image.h $DESTDIR/x11/include/SDL
install -d $DESTDIR/x11/lib
install .libs/libSDL_image-1.2.so.0.* $DESTDIR/x11/lib/libSDL_image-1.2.so.0
ln -sf libSDL_image-1.2.so.0 $DESTDIR/x11/lib/libSDL_image.so
install -d $DESTDIR/x11/lib/pkgconfig
install -m 644 SDL_image.pc $DESTDIR/x11/lib/pkgconfig

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/sdl_image-build.pc << EOF
Name: SDL_image
Version: 1.2.12
Description: Image loading library for SDL
Requires: $REQUIRES

devel=\\
/x11/include/SDL/SDL_image.h \\
/x11/lib/libSDL_image.so \\
/x11/lib/pkgconfig/SDL_image.pc

exec=\\
/x11/bin/showimage \\
/x11/lib/libSDL_image-1.2.so.0
EOF

exit
--- SDL_image-1.2.12.orig/configure.in
+++ SDL_image-1.2.12/configure.in
@@ -1,5 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(README)
+AC_INIT(SDL_image, 1.2.12)
+AC_CONFIG_SRCDIR(README)
+AC_CONFIG_MACRO_DIR(acinclude)
 
 dnl Set various version strings - taken gratefully from the GTk sources
 
@@ -42,7 +44,7 @@
 AC_CANONICAL_HOST
 
 dnl Setup for automake
-AM_INIT_AUTOMAKE(SDL_image, $VERSION)
+AM_INIT_AUTOMAKE(foreign)
 
 dnl Check for tools
 AC_PROG_LIBTOOL
